lib/net.c

```diff
@@ -1200,12 +1201,119 @@ start_iface(struct gatekeeper_if *iface, unsigned int num_attempts_link_get)
 	return ret;
 }
 
+static void
+calculate_net_config_para(struct net_config *net_conf)
+{
+	int i;
+
+	/* The total number of RX queues on any interface. */
+	uint16_t gatekeeper_num_rx_queues = net_conf->front.num_rx_queues +
+		(net_conf->back_iface_enabled ?
+		 net_conf->back.num_rx_queues : 0);
+
+	/* The total number of TX queues on any interface. */
+	uint16_t gatekeeper_num_tx_queues = net_conf->front.num_tx_queues +
+		(net_conf->back_iface_enabled ?
+		 net_conf->back.num_tx_queues : 0);
+
+	const char *filtered_blocks[] = { "dynamic_conf" };
+
+	/*
+	 * The total number of lcores used by any functional block
+	 * for either RX or TX except for Dynamic Configuration.
+	 */
+	int gatekeeper_num_lcores = launch_count_lcores(filtered_blocks,
+		RTE_DIM(filtered_blocks));
+
+	/*
+	 * The maximum number of receive descriptors to
+	 * allocate for the receive ring over all interfaces.
+	 */
+	uint16_t gatekeeper_num_rx_desc =
+		net_conf->front.gatekeeper_num_rx_desc;
+
+	/*
+	 * The maximum number of transmit descriptors to
+	 * allocate for the transmit ring over all interfaces.
+	 */
+	uint16_t gatekeeper_num_tx_desc =
+		net_conf->front.gatekeeper_num_tx_desc;
+
+	if (net_conf->back_iface_enabled) {
+		gatekeeper_num_rx_desc = RTE_MAX(gatekeeper_num_rx_desc,
+			net_conf->back.gatekeeper_num_rx_desc);
+
+		gatekeeper_num_tx_desc = RTE_MAX(gatekeeper_num_tx_desc,
+			net_conf->back.gatekeeper_num_tx_desc);
+	}
+
+	/*
+	 * The number of elements in the mbuf pool.
+	 *
+	 * The optimum size (in terms of memory usage) for a mempool is when
+	 * it is a power of two minus one.
+	 *
+	 * Need to provision enough memory for the worst case. According to
+	 * l3fwd sample application, it's essentially the number of RX
```

Comment:
Replace `According to l3fwd sample application, it's essentially the number of RX` with `It's the number of RX`.